src/text-editor-presenter.coffee

```diff
@@ -564,6 +644,18 @@ class TextEditorPresenter
       decorationClasses.push(decoration.getProperties().class)
     decorationClasses
 
+  # Returns a {Set} of {Decoration}s on the given custom gutter from startRow to endRow (inclusive).
+  customGutterDecorationsInRange: (gutterName, startRow, endRow) ->
+    decorations = new Set
+
+    return decorations if @model.isMini() or gutterName is 'line-number' or
+      not @customGutterDecorationsByGutterNameAndScreenRow[gutterName]
+
+    for bufferRow in @model.bufferRowsForScreenRows(@startRow, @endRow - 1)
```

Comment:
addressed
for now, there's a caveat that the decorations won't react to folding, due to a bug that makes the presenter unaware of when markers change screen rows